rcl/src/rcl/subscription.c

```diff
@@ -186,13 +186,19 @@ rcl_subscription_get_rmw_handle(const rcl_subscription_t * subscription)
 bool
 rcl_subscription_is_valid(const rcl_subscription_t * subscription)
 {
+  const rcl_subscription_options_t * options = rcl_subscription_get_options(subscription);
```

Comment:
You could split `rcl_subscription_get_options` up like this:

```c
return_type
_subscription_get_options(params...) {
  // just return the options, no checks
  return options
}

return_type
rcl_subscription_get_options(params....) {
   // do checks, like call is_valid
   return _subscription_get_options(params...);
}
```

Then you could use the internal `_subscription_get_options` function in `is_valid`.

Or since it is so simple you could just duplicate the logic to extract the options in both places.